src/pages/index.js

```diff
@@ -1,93 +1,236 @@
 const addModal = "#add-modal";
+const editProfileImgForm = document.forms["edit-profile-image-form"];
 const addCardForm = document.forms["add-card-form"];
 const previewModal = "#preview-modal";
+const deleteModal = "#delete-modal";
 
-//add card section to DOM & add initial cards
+const api = new Api({
+  baseUrl: apiConfig.baseUrl,
+  headers: apiConfig.headers,
+});
 
-function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", imageClick);
-  return card.getView();
-}
+// Create user info instance
+const userInfo = new UserInfo({
+  userName: profileName,
+  userJob: profileDescription,
+});
+
+//receive user info
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      userName: res.name,
+      userJob: res.about,
+    });
+
+    profileImageEl.setAttribute("src", res.avatar);
+  })
+  .catch((err) => {
+    console.error("Failed to receive user info", err);
+  });
+
+//add card section to DOM & add initial cards
 
 const cardSection = new Section(
   {
-    items: initialCards,
-    renderer: (cardData) => {
+    renderer: (cardData, method) => {
+      if (cardData.link) {
+        cardData.cardLink = cardData.link;
+      }
+      if (cardData.name) {
+        cardData.cardName = cardData.name;
+      }
+
       const cardElement = createCard(cardData);
-      cardSection.addItem(cardElement, "append");
+      cardSection.addItem(cardElement, method);
     },
   },
   ".cards__list"
 );
 
-cardSection.renderItems();
+api
+  .getInitialCards()
+  .then((cards) => {
+    cardSection.renderItems(cards, "append");
+  })
+  .catch((err) => {
+    console.error("Failed to receive card data", err);
+  });
+
+function createCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    imageClick,
+    deleteClick,
+    likeClick
+  );
+  return card.getView();
+}
 
 //Submit handling functions
 
-function handleProfileFormSubmit(input) {
-  userInfo.setUserInfo(input);
-  profileEditPopup.close();
+function handleProfileFormSubmit(input, button) {
+  api
+    .setUserInfo(input)
+    .then((data) =>
+      userInfo.setUserInfo({ userName: data.name, userJob: data.about })
+    )
+    .catch((err) => {
+      console.error("Failed to save changes in user info", err);
+    })
+    .finally(() => {
+      profileEditPopup.close();
+      button.textContent = "Save";
+    });
 }
 
-function handleAddCardFormSubmit(input) {
-  const cardElement = createCard(input);
-  cardSection.addItem(cardElement, "prepend");
-  addCardPopup.close();
-  addCardForm.reset();
-  formValidators["add-card-form"].disableButton();
+function handleProfileImgFormSubmit(input, button) {
+  api
+    .editProfileImg(input.imageLink)
+    .then((data) => {
+      if (data && data.avatar) {
+        profileImageEl.src = data.avatar;
+      }
+    })
+    .catch((err) => {
+      console.error("Failed to update profile image:", err);
+    })
+    .finally(() => {
+      profileEditImagePopup.close();
```

Comment:
```js

      profileEditImagePopup.close();
      editProfileImgForm.reset();
      formValidators["edit-profile-image-form"].disableButton();
```

These 3 lines of code should be in `then` in line 119

--------------------

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.

Reply:
You need to **disable** the button only in a `then` block after a **successful response** to be able to click it again if there is a server error.

Reply:
**Please, correct it everywhere in the code**